### DISCUSSION TITLE
Not sure how to use Placeholder vs Default Value pre-fill from State

### DISCUSSION THREAD

**Discussion by assia-zoubir** [Budibase/budibase#13713] [Help] [answered]
Hello , i'm a bit of a bind.
In one page of my application, i execute a query, then use the data from the result to fill a form in another page.
<img width="436" alt="image" src="https://github.com/Budibase/budibase/assets/149948208/3bb98edd-69d0-439c-a924-33e8efea77ce">
<img width="669" alt="image" src="https://github.com/Budibase/budibase/assets/149948208/34784022-7397-4e00-8e60-8095fd2eb2b0">

In this form, If i don't set the default value and only the placeholder with the data from the state, the page works normally, and the form appears to be pre-filled, however, if i pull the data from the fields of the form and execute a query with them, the fields i haven't changed and are still the placeholders will be sent as null (or 2:00:00 for the datetime)

if i do change the default value , then , the page also appears to be working normally, and the query pulls from the fields correctly, but the timepicker becomes almost unusable, because it always add +1 to the hour that i input in, if i change the value whatsoever (minute or hour), an issue that is otherwise not present. what's more it appears that the datetime is actually sent with one more hour than what is displayed , after checking the database after the query
here is a video of the bug : [video](https://streamable.com/n2t8ib)

I don't know where to go from there or if there is another way to do what I'm trying to do ..

Thank you in advance.

**Comment by mikesealey** [comment, marked as answer]
Hi @assia-zoubir

Your video doesn't seem to load, but I've got some suggestions that might help.

When you run your query from elsewhere in your app, set the value in state [Documentation about State here](https://docs.budibase.com/docs/app-state)

Then for debugging purposes, create a paragraph in your app that contains what you're setting the placeholder value to be. You may find that you're not quite getting back what you expect and prop-drilling isn't working. If that's the case, cut off the last piece of the variable 
eg:
`JSON.parse($(State.current_task)).data[0].start_time` will become `JSON.parse($(State.current_task)).data[0`

And keep working back until you get something you recognise.

Once you've found the right variables and properly indexed them, you should check to see if there's a value there when you set the placeholder text. If you try to return the value of a key that doesn't exist, Budibase will throw an error. Consider using Optional Chaining to avoid getting these errors. You can learn more about it here: [MDN Optional Chaining.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

I hope this helps get you unstuck - if you haven't already, consider joining the [Budibase Discord Server](https://discord.com/invite/ZepTmGbtfF).
